x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java

```diff
@@ -504,29 +508,39 @@ public void testPauseAndResumeWithMultipleAutoFollowPatterns() throws Exception
                     }
                 } catch (Exception e) {
                     throw new AssertionError(e);
+                } finally {
```

Comment:
If we throw an exception on an iteration before the 9th then this thread will terminate and `latchNine.await()` will get stuck, so I don't think the `finally` really helps us here. Could we instead use a timeout on the `await()` calls? Something more generous than the `assertBusy()` default of 10 seconds ofc.

Reply:
Something like 1687096ffff8f84569634303e83d6f4109e023a6 ?